doc/nrf/protocols/matter/end_product/configuring_cd.rst

```diff
@@ -0,0 +1,83 @@
+Given that the current software version can change with updates to the device firmware, the CSA verifies if the new software version is matching the previous conditions for granting the certification.
+If it does, the CSA regenerates CD for the new software version.
+If it does not, you must apply for a new certification.
+
+Because CD can change, it cannot be stored in the factory data set or other parts of firmware that cannot be modified during the device lifetime.
+For this reason, it is included in the part of firmware that you can modify after the certification.
+
+Modifying Certification Declaration
+***********************************
+
+In the |NCS| implementation of Matter, you can configure CD by defining the :c:macro:`CHIP_DEVICE_CONFIG_CERTIFICATION_DECLARATION` define in the :file:`chip_project_config.h` file.
+You can locate the array of bytes related to CD by running the search with the following condition:
+
+.. code-block:: console
+
+   cat CD.der | xxd -I
+
+Storing Certification Declaration in Zephyr
+===========================================
+
+Alternatively, you can opt for storing CD in the Zephyr's Settings subsystem, which allows for storing data even after the device has been programmed.
+For example, this lets you add CD to the subsystem through CLI, or use Zephyr's API to store CD in the Settings subsystem within the code.
+
+To enable this configuration method, set the :kconfig:option:`CONFIG_CHIP_CERTIFICATION_DECLARATION_STORAGE` Kconfig option in the :file:`prj.conf` file instead of the define in :file:`chip_project_config.h`.
+
+.. _ug_matter_device_configuring_cd_generating_steps:
+
+Generating Certification Declaration
+************************************
+
+You can generate CD for your end product integration testing or as part of the official regeneration procedure with the CSA.
+
+Generating CD for integration testing
+=====================================
+
+To generate CD for integration testing, complete the following steps:
+
+1. Install the chip-cert tool.
+   This is a Matter tool used for generating new certificates.
+   Complete the following steps:
+
+   a. Navigate to the :file:`connectedhomeip` root directory.
+   a. In a terminal, run the following command to build the tool executable file:
+
+      .. code-block:: console
+
+         cd src/tools/chip-cert && gn gen out && ninja -C out chip-cert
+
+1. Run the following command to generate CD:
+
+   .. code-block:: console
+
+      chip-cert gen-cd --key credentials/test/certification-declaration/Chip-Test-CD-Signing-Key.pem --cert credentials/test/certification-declaration/Chip-Test-CD-Signing-Cert.pem --out CD.der --format-version 1 --vendor-id 0xFFF1 --product-id 0x8006 --device-type-id 0xA --certificate-id ZIG20142ZB330003-24 --security-level 0 --security-info 0 --certification-type 1 --version-number 0x2694
```

Comment:
user should change the `vendor-id` and `product-id` for sure.
regarding the paths to test signing key and test signing cert, it can be found in `modules/lib/matter/credentials/test/certification-declaration/`